src/core/index.ts

```diff
@@ -396,6 +427,10 @@ export class ZarrArray {
     return new DTYPE_TYPEDARRAY_MAPPING[this.dtype](buffer);
   }
 
+  private toRawArray(buffer: Buffer | ArrayBuffer) {
```

Comment:
I was a bit confused at first how we can just assume the chunksize or dtype here, but we only ever call this with a whole chunk, then it makes sense. Maybe this can be `chunkToRawArray`? or `chunkBufferToRawArray` or something like that?

Reply:
Good idea. I've been thinking of changing the `out.set()` function within `ZarrArray.getChunkItem` to `out.setFromChunkItem` because it's a very specific task of decoding a full chunk (with its selection) directly to the output array (which I doubt users will be trying to do directly). 

This seems distinct from what would be expected with `RawArray.set(typedArray, selection)`, where the typed array must be the exact size of the selection, and the indexing of `typedArray` can be inferred from the selection itself.